src/ESLint.Formatter/test/sarif-with-rules.js

```diff
@@ -0,0 +1,320 @@
+/**
+ * @fileoverview Tests for SARIF format.
+ * @author Chris Meyer
+ */
+
+"use strict";
+
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const assert = require("chai").assert,
+    formatter = require("../../../lib/formatters/sarif");
+
+
+//------------------------------------------------------------------------------
+// Global Test Content
+//------------------------------------------------------------------------------
+
+const rules = new Map();
+
+rules.set("no-unused-vars", {
+    meta: {
+        type: "suggestion",
+        docs: {
+            description: "disallow unused variables",
+            category: "Variables",
+            recommended: true,
+            url: "https://eslint.org/docs/rules/no-unused-vars"
+        },
+        fixable: "code"
+    }
+});
+rules.set("no-extra-semi", {
+    meta: {
+        type: "suggestion",
+
+        docs: {
+            description: "disallow unnecessary semicolons",
+            category: "Possible Errors",
+            recommended: true,
+            url: "https://eslint.org/docs/rules/no-extra-semi"
+        },
+
+        fixable: "code",
+        schema: [],
+
+        messages: {
+            unexpected: "Unnecessary semicolon."
+        }
+    }
+});
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+describe("formatter:sarif", () => {
+    describe("when passed no messages", () => {
+        const sourceFilePath = "foo.js";
+        const code = [{
+            filePath: sourceFilePath,
+            messages: []
+        }];
+
+        it("should return a log with one file and no results", () => {
+            const result = JSON.parse(formatter(code, null));
+
+            assert.hasAllKeys(result.runs[0].files, sourceFilePath);
+            assert.strictEqual(result.runs[0].files[sourceFilePath].fileLocation.uri, sourceFilePath);
+            assert.isUndefined(result.runs[0].results);
+        });
+    });
+});
+
+describe("formatter:sarif", () => {
+    describe("when passed one message", () => {
+        const sourceFilePath = "foo.js";
```

Comment:
>foo [](start = 32, length = 3)

Can you replace all uses of term 'foo' with something else? For now and always. :)

Reply:
Sorry, dumb

---
In reply to: [249142994](https://github.com/Microsoft/sarif-sdk/pull/1222#discussion_r249142994) [](ancestors = 249142994)